optuna/integration/_lightgbm_tuner/sklearn.py

```diff
@@ -1,49 +1,887 @@
+import pathlib
 from typing import Any
-import warnings
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
 
 import lightgbm as lgb
+import numpy as np
+import pandas as pd
+from scipy.sparse import spmatrix
+from sklearn.base import BaseEstimator
+from sklearn.base import ClassifierMixin
+from sklearn.base import is_classifier
+from sklearn.base import RegressorMixin
+from sklearn.preprocessing import LabelEncoder
+from sklearn.utils import check_array
+from sklearn.utils import check_consistent_length
+from sklearn.utils.class_weight import compute_sample_weight
+from sklearn.utils.multiclass import check_classification_targets
+from sklearn.utils.validation import _assert_all_finite
+from sklearn.utils.validation import _num_samples
+from sklearn.utils.validation import column_or_1d
 
+from optuna import logging
+from optuna import study as study_module
+from optuna.integration._lightgbm_tuner import optimize
 
-class LGBMModel(lgb.LGBMModel):
-    """Proxy of lightgbm.LGBMModel.
 
-    See: `pydoc lightgbm.LGBMModel`
+if lgb.__version__ >= "2.3":
+    from lightgbm.sklearn import _EvalFunctionWrapper
+    from lightgbm.sklearn import _ObjectiveFunctionWrapper
+else:
+    from lightgbm.sklearn import _eval_function_wrapper as _EvalFunctionWrapper
+    from lightgbm.sklearn import _objective_function_wrapper as _ObjectiveFunctionWrapper
+
+__all__ = ["LGBMModel", "LGBMClassifier", "LGBMRegressor"]
+
+OneDimArrayLikeType = Union[np.ndarray, pd.Series]
+TwoDimArrayLikeType = Union[np.ndarray, pd.DataFrame, spmatrix]
+
+RandomStateType = Union[int, np.random.RandomState]
+
+OBJECTIVE2METRIC = {
+    # classification
+    "binary": "binary_logloss",
+    "multiclass": "multi_logloss",
+    "softmax": "multi_logloss",
+    "multiclassova": "multi_logloss",
+    "multiclass_ova": "multi_logloss",
+    "ova": "multi_logloss",
+    "ovr": "multi_logloss",
+    # regression
+    "mean_absoluter_error": "l1",
+    "mae": "l1",
+    "regression_l1": "l1",
+    "l2_root": "l2",
+    "mean_squared_error": "l2",
+    "mse": "l2",
+    "regression": "l2",
+    "regression_l2": "l2",
+    "root_mean_squared_error": "l2",
+    "rmse": "l2",
+    "huber": "huber",
+    "fair": "fair",
+    "poisson": "poisson",
+    "quantile": "quantile",
+    "mean_absolute_percentage_error": "mape",
+    "mape": "mape",
+    "gamma": "gamma",
+    "tweedie": "tweedie",
+}
```

Comment:
IMO, we can reuse `_ALIAS_METRIC_LIST` in [`optuna/integration/_lightgbm_tuner/alias.py`](https://github.com/optuna/optuna/blob/0e9557a0930755127b96613d1aacfc652eeb37f3/optuna/integration/_lightgbm_tuner/alias.py#L43-L104). What do you think?

Reply:
Fixed in a5db770.

Reply:
https://github.com/optuna/optuna/pull/1018/checks?check_run_id=1230177914
Sorry, I don't know how to fix the CI error.